### PR TITLE
Updates to allow for bucket policies and provisioning of dependencies

### PR DESCRIPTION
**Pull request by frasercrichton**
Added startup scripts to add some of the dependencies Ansible adds to speed up the Ansible scripts. Also updated the policies on the buckets so they are read-only to the public.

### CODE REVIEW

File: provision/README.md [hunk 1]
```diff
@@ -67,3 +67,9 @@ When the scripts run they create an 'ephemeral' external IP address for the VM b
 You can rerun Terraform to change infrastructure at any time and you can also destroy all the Timemap related infrastructure at any time.
 
 Your infrastructure is now ready to be provisioned with Timemap and Datasheet as described in the deploy_timemap [README.md](../../deploy_timemap/README.md)
+
+## Storage Buckets
+
+With both AWS and GCP we have included provisioning of storage buckets so you can store the images, video and documents that Timemap relies on. Once you have provisioned the infrastructure you can upload these resources to the buckets and reference them in your spreadsheets.
```
Comment (breezykermo) [inline provision/README.md:73]: Nice. Maybe "and reference the appropriate URLs in your datasheet-server spreadsheets".

File: provision/README.md [hunk 2]
```diff
@@ -67,3 +67,9 @@ When the scripts run they create an 'ephemeral' external IP address for the VM b
 You can rerun Terraform to change infrastructure at any time and you can also destroy all the Timemap related infrastructure at any time.
 
 Your infrastructure is now ready to be provisioned with Timemap and Datasheet as described in the deploy_timemap [README.md](../../deploy_timemap/README.md)
+
+## Storage Buckets
+
+With both AWS and GCP we have included provisioning of storage buckets so you can store the images, video and documents that Timemap relies on. Once you have provisioned the infrastructure you can upload these resources to the buckets and reference them in your spreadsheets. 
+
+You can upload files through the UI in the console or use the command line and CLI. For details of how to upload resources with the CLI see the [aws-account-readme.md](docs/aws-account-readme.md) or the [gcp-account-readme.md](docs/gcp-account-readme.md).
```
Comment (breezykermo) [inline provision/README.md:75]: These links are broken for me (and also the ones above in this file). I think they should be prefixed with `docs/`.

File: provision/aws-instance/aws-forensic-architecture.tf [hunk 3]
```diff
@@ -1,60 +1,53 @@
-variable "region" {
```
Comment (breezykermo) [inline provision/aws-instance/aws-forensic-architecture.tf:1]: Just a thought- would it make more sense to call this file 'aws-timemap.tf'? And the same with the corresponding .tf file in gcp.

File: provision/aws-instance/aws-forensic-architecture.tf [hunk 4]
```diff
@@ -1,60 +1,53 @@
...
-variable "connection_user" {
-  default = "ubuntu"
-}
-
 provider "aws" {
   profile = "default"
   region  = var.region
 }
 
 resource "aws_s3_bucket" "forensic-architecture-bucket" {
```
Comment (breezykermo) [inline provision/aws-instance/aws-forensic-architecture.tf:6]: similarly here- I think we can call these resources 'timemap-bucket'.

File: provision/aws-instance/aws-forensic-architecture.tf [hunk 5]
```diff
@@ -1,60 +1,53 @@
...
-  # NOTE: S3 bucket names must be unique across _all_ AWS accounts, so
-  # this name must be changed before applying this example to avoid naming
-  # conflicts.
   bucket = var.bucket-name
-  acl    = "private"
   region = var.region
 }
 
+resource "aws_s3_bucket_policy" "forensic-architecture-bucket-policy" {
```
Comment (breezykermo) [inline provision/aws-instance/aws-forensic-architecture.tf:11]: and here: --> 'timemap-bucket-policy'

File: provision/aws-instance/aws-forensic-architecture.tf [hunk 6]
```diff
@@ -1,60 +1,53 @@
...
+    sid    = "PublicRead"
+    effect = "Allow"
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
+    }
+    actions   = ["s3:GetObject"]
+    resources = [format("arn:aws:s3:::%s/*", var.bucket-name)]
+  }
+}
```
Comment (breezykermo) [inline provision/aws-instance/aws-forensic-architecture.tf:28]: beaut

File: provision/aws-instance/aws-forensic-architecture.tf [hunk 7]
```diff
@@ -64,9 +57,9 @@ resource "aws_eip" "EC2-IP-address" {
   depends_on = [aws_instance.forensic-architecture-vm]
 }
 
-resource "aws_security_group" "forensic-architecture-web-node" {
-  name        = "forensic-architecture-web-node"
-  description = "Web Security Group"
+resource "aws_security_group" "forensic-architecture-security-group" {
+  name        = "forensic-architecture-security-group"
+  description = "Web Security Group for Forensic Architecture EC2"
```
Comment (breezykermo) [inline provision/aws-instance/aws-forensic-architecture.tf:62]: "Timemap EC2"

File: provision/docs/aws-account-readme.md [hunk 8]
```diff
@@ -30,3 +30,25 @@ The AWS CLI needs to be configured with security credentials so it can connect t
...
+aws s3 cp your-directory-name s3://forensic-architecture-bucket --recursive
+```
+
+### Sync
+
+To synchronise everything in the current directory (upload and download anything already there) run the following command:
+
+```
+aws s3 sync . s3://forensic-architecture-bucket
+```
```
Comment (breezykermo) [inline provision/docs/aws-account-readme.md:52]: This is a great way to manage the resources; thanks so much for PR-ing!

File: provision/gcp-instance/gcp-forensic-architecture.tf [hunk 9]
```diff
@@ -1,32 +1,28 @@
-variable "region" {
```
Comment (breezykermo) [inline provision/gcp-instance/gcp-forensic-architecture.tf:1]: Same thing here (see comments above): from 'forensic-architecture' --> 'timemap'

File: provision/gcp-instance/variables.tf [hunk 10]
```diff
@@ -0,0 +1,20 @@
+variable "region" {
```
Comment (breezykermo) [inline provision/gcp-instance/variables.tf:1]: 'forensic-architecture' --> 'timemap' here as well.

File: provision/docs/aws-account-readme.md (hunk 8 above)
Comment (frasercrichton) [reply provision/docs/aws-account-readme.md:52]: No worries :)

**Comment by frasercrichton** [comment]
Think this should be good to go now @breezykermo.

**Pull request merged**